datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/refresh.py

```diff
@@ -0,0 +1,63 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+import threading
+import time
+from collections import defaultdict
+
+import pywintypes
+import win32pdh
+
+from ....utils.time import get_precise_time
+
+
+class WindowsPerformanceObjectRefresher(threading.Thread):
+    INTERVAL = 60
+
+    def __init__(self):
+        name = self.__class__.__name__
+        super(WindowsPerformanceObjectRefresher, self).__init__(name=name)
+
+        self.logger = logging.getLogger(name)
+        self.last_refresh = {}
+        self.servers = defaultdict(int)
+
+    def run(self):
+        # Refresh the list of performance objects for every server, see:
+        # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhenumobjectitemsa#remarks
+        #
+        # We do this in a separate thread to avoid internal lock contention of Windows APIs
+        while True:
+            # Agent is shutting down
+            if not any(self.servers.values()):
+                return
+
+            for server, count in self.servers.items():
+                if not count:
+                    continue
+
+                now = get_precise_time()
+                if server in self.last_refresh and now - self.last_refresh[server] < self.INTERVAL:
```

Comment:
What if someone or something will change the time AFTER we memorized last enum time and make current way earlier it will disabled enum until that delta passed which may be a while in some cases. I think you need to make an extra validation check,

Reply:
Ok, let's presume that on one iteration after enum the content of self.last_refresh[server] is 6:00:00 pm. Then around 6:00:10 pm a time service or a user change current time (there are few reasons for that) to 9:00:10 am. It will mean

`if server in self.last_refresh and now - self.last_refresh[server] < self.INTERVAL:` will succeed and skip enum for 9 more hours until new clock will catchup with the value of old clock. I was suggesting of additional check and resetting self.last_refresh[server] if that happen

Reply:
That would not affect this https://docs.python.org/3/library/time.html#time.perf_counter